src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -608,14 +606,14 @@ def execute(self) -> Sequence[QueryResult]:
             return []
 
         with metrics.timer(key="ddm.metrics_api.execution.total_execution_time"):
-            results = self._bulk_execute()
+            self._execution_loop()
 
         metrics.distribution(
             key="ddm.metrics_api.execution.number_of_executed_queries",
             value=self._number_of_executed_queries,
         )
 
-        return results
+        return cast(Sequence[QueryResult], self._query_results)
```

Comment:
We assume that when we are here, only query results will be there. If this assumption won't hold, we will have exceptions down the line, so it will still be easy to spot and fix.